lib/Sema/TypeCheckAttr.cpp

```diff
@@ -1932,7 +1932,7 @@ void AttributeChecker::visitMainTypeAttr(MainTypeAttr *attr) {
   stmts.push_back(returnStmt);
   auto *body = BraceStmt::create(context, SourceLoc(), stmts,
                                 SourceLoc(), /*Implicit*/true);
-  func->setBodyParsed(body);
+  func->setBody(body, AbstractFunctionDecl::BodyKind::TypeChecked);
```

Comment:
This is still not enough. ASTVisitor says:
```
unspecialized reference with polymorphic type <Self where Self : OtherThing, Self : Runnable> (Self.Type) -> () -> ()
(declref_expr implicit type='<Self where Self : OtherThing, Self : Runnable> (Self.Type) -> () -> ()' location=/Users/rintaro/Repositories/swift-oss/swift/test/attr/ApplicationMain/attr_main_struct_from_two_protocols_one_missing.swift:18:1 range=[/Users/rintaro/Repositories/swift-oss/swift/test/attr/ApplicationMain/attr_main_struct_from_two_protocols_one_missing.swift:18:1 - line:18:1] decl=attr_main_struct_from_two_protocols_one_missing.(file).Runnable extension.main()@/Users/rintaro/Repositories/swift-oss/swift/test/attr/ApplicationMain/attr_main_struct_from_two_protocols_one_missing.swift:12:17 [with (substitution_map generic_signature=<Self where Self : OtherThing, Self : Runnable> (substitution Self -> EntryPoint))] function_ref=unapplied)
```

Reply:
Also, setting `BodyKind::TypeChecked` causes:
```
/Users/rintaro/Repositories/swift-oss/swift/test/attr/ApplicationMain/attr_main_struct_available_future.swift:5:10: error: expected error not produced
@main // expected-error {{'main()' is only available in macOS 10.99 or newer}}
     ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
It will take a bit more work to build a fully type checked AST here, for example you have to set types on all the Exprs. Maybe it’s best to avoid doing that for now?

Reply:
The specific verifier error is because you have a DeclRefExpr without substitutions. If you want to pursue building fully checked AST here you’ll need to provide a substitution map.

Reply:
Removed `setType()`/`setInterfaceType()` part from this function. It should be done in constraint system.